Source/TimeIntegration/StrainRate.cpp

```diff
@@ -136,3 +134,93 @@ ComputeStrainRate(const int &i, const int &j, const int &k,
 
   return strainRate;
 }
+
+AMREX_GPU_DEVICE
+Real
+ComputeExpansionRate(const int &i, const int &j, const int &k,
+                     const Array4<Real>& u, const Array4<Real>& v, const Array4<Real>& w,
+                     const enum NextOrPrev &nextOrPrev,
+                     const enum MomentumEqn &momentumEqn,
+                     const enum DiffusionDir &diffDir,
+                     const GpuArray<Real, AMREX_SPACEDIM>& cellSize) {
+    Real dx = cellSize[0];
+    Real dy = cellSize[1];
+    Real dz = cellSize[2];
+
+    Real expansionRate = 0;
+
+    //TODO: Check if it is more efficient to store computed divergence at cell-centers rather than
```

Comment:
I'm pretty sure it will be more efficient to precompute the divergence. Right now it takes 6 divergence computations to compute a single stress, so precomputing the divergence should reduce the cost associated with calls to this function by 6x. I think that can wait though. There are several places that could be similarly revised.